pennylane/transforms/dynamic_one_shot.py

```diff
@@ -195,26 +206,35 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
                 new_measurements.append(SampleMP(obs=m.obs))
             else:
                 new_measurements.append(m)
-    for op in circuit:
-        if is_mcm(op):
+    new_operations = []
+    for op in circuit.operations:
+        if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
+            new_op = op
+            new_op.bypass_postselect = True
+            new_operations.append(new_op)
```

Comment:
I think I mentioned in some other pr, but in this case `new_op` is just a different name for the same object. We should either copy the op or remove the `new_op` reference.

Reply:
Will remove following the introduction of `postselection_mode`.